File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollars;
    int k = 0; //k means nr of coins used
    
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);
    
    int cents = round(dollars * 100); //transforms $ in cents
    
    while (cents >= 25)
    {
        cents = cents - 25;
        k++;
    }
        
    while (cents >= 10)
    {
        cents -= 10;
        k++;
    }
        
    while (cents >= 5)
    {
        cents -= 5;
        k++;
    }
    
    while (cents >= 1)
    {
        cents -= 1;
        k++;
    }
    
    printf("%i\n", k);
}

File: cash.py
from cs50 import get_float

k = 0
while True:
    dollars = get_float("Change owed: ")
    if dollars >= 0:
        break
    
cents = round(dollars * 100)
while cents >= 25:
    cents -= 25
    k += 1
    
while cents >= 10:
    cents -= 10
    k += 1
    
while cents >= 5:
    cents -= 5
    k += 1
    
while cents >= 1:
    cents -= 1
    k += 1
    
print (k)
    